File: 8 - Dictionaries and Maps/Solution.swift
// Enter your code here. Read input from STDIN. Print output to STDOUT
import Foundation
let n = Int(readLine()!)!
var phonebook = [String:String]()

for _ in 0...n-1 {
    let contact = readLine()!
    let person = contact.components(separatedBy: " ")
    phonebook[person[0]] = person[1]
    
}


var queryName = readLine()
while queryName != nil {
    if phonebook.keys.contains(queryName!){
        print("\(queryName!)=\(phonebook[queryName!]!)")
    }else{
        print("Not found")
    }
    queryName = readLine()
}

File: 8 - Dictionaries and Maps/Solution.cpp
#include <map>
#include <iostream>

using namespace std;

int main() {
    int N;
    string name;
    cin >> N;

    map<string, int> phone_book;

    for (int i = 0; i < N; i++) {
        cin >> name;

        if (!phone_book[name]) {
            cin >> phone_book[name];
        }
    }

    while (cin>>name) {

        if (phone_book[name]) {
            cout << name << "=" << phone_book[name] << endl;
        } else {
            cout << "Not found" << endl;
        }
    }

    return 0;
}
